projects/demo/src/app/navigation/navigation/navigation.component.ts

```diff
@@ -0,0 +1,139 @@
+import { Component, OnInit, ViewChild, Inject } from '@angular/core';
+import { NavigationComponent, NavigationData } from '../../../../../../src/public_api';
+import { NavigationController } from 'src/app/ui/navigation/navigation-controller.service';
+
+const example = `
+// html
+
+<iw-navigation #nav>
+  <h1>Here the content of the first view</h1>
+</iw-navigation>
+
+<button (click)="present()">Present</button>
+
+// ts: YourComponent
+
+import { NavigationComponent, NavigationData } from 'iwerk-angular-ui';
+
+@ViewChild('nav') nav: NavigationComponent;
+
+click() {
+  this.nav.present(SomeComponent, { someCustomData })
+    .then(someValueDismissedByThePresentedView => {
+      …
+    });
+}
+
+// ts: SomeComponent
+
+constructor(private navCtrl: NavigationController) {
```

Comment:
@astme in terms of developer experience, wouldn't it be even better to directly get the navigation component instead of another nav ctrl service?

Meaning that, from a stacked component, you could do the following:

```
constructor(private navigation: NavigationComponent) {…}
```

Reply:
@kevinmerckx I suppose most of the developers expect to see a service instead of injecting a parent component directly.